File: LifecycleSeekbarRatingbarDate/app/src/main/java/com/example/taling6/MainActivity.java
package com.example.taling6;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

// 안드로이드 생명주기 (액티비티 생명주기)
// 없어도 앱은 만들수있음
// 제대로된 앱(최적화, 기능분리)을 만들기위해서는 필수
// 액티비티 생명주기 개념이 없는 앱은 99.9999999999999% 없다.

// 로그캣(디버깅, 어디서틀렸고)

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

//        System.out.println("나는 onCreate입니다");
        Log.i("MY_LOG", "나는 onCreate입니다");
    }

    // 이클립스 (알트 쉬프트 s)
    // 안드로이드 스튜디오 (컨트롤 o)

    @Override
    protected void onStart() {
        super.onStart();
        Log.i("My_LOG", "나는 onStart");

        // 로그

        //daskldjaskld오류 오류코드

        //로그
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i("My_LOG", "나는 onResume입니다");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i("My_LOG", "나는 onPause입니다");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i("My_LOG", "나는 onStop입니다");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i("My_LOG","나는 onDestroy입니다");
    }
}

File: LifeCycle/app/src/main/java/com/puresoftware/lifecycle/MainActivity.kt
package com.puresoftware.lifecycle

import android.annotation.SuppressLint
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.widget.Button

class MainActivity : AppCompatActivity() {

    val TAG = MainActivity::class.java.simpleName

    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btnGoSub = findViewById<Button>(R.id.btn_go_sub)

        var intent:Intent = Intent(this,Subactivity::class.java)

        btnGoSub.setOnClickListener {
            startActivity(intent)
        }

    }

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)

        Log.i(TAG,"onCreate 실행")
    }

    override fun onStart() {
        super.onStart()
        Log.i(TAG,"onStart 실행")
    }

    override fun onResume() {
        super.onResume()
        Log.i(TAG,"onResume 실행")
    }

    override fun onPause() {
        super.onPause()
        Log.i(TAG,"onPause 실행")
    }

    override fun onRestart() {
        super.onRestart()
        Log.i(TAG,"onRestart 실행")
    }

    override fun onStop() {
        super.onStop()
        Log.i(TAG,"onStop 실행")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i(TAG,"onDestroy 실행")
    }
}